lib/action_controller/serialization.rb

```diff
@@ -25,6 +25,11 @@ def get_serializer(resource, options = {})
           "Please pass 'adapter: false' or see ActiveSupport::SerializableResource#serialize"
         options[:adapter] = false
       end
+      if resource.respond_to?(:current_page) && resource.respond_to?(:total_pages)
+        options[:pagination] = {}
+        options[:pagination][:original_url] = original_url
+        options[:pagination][:query_parameters] = query_parameters
+      end
```

Comment:
There's a bunch wrong in the specifics of the code above, I totally admit. I was more interested in the flow.. but e.g.

```ruby
def serializable_hash(options)
  pagination_links = {}
```

is wrong.  those options passed in to serializable_hash should be the pagination_links options.  The thing that has the context should be passed into the initializer, I think.